tuftool/src/source.rs

```diff
@@ -58,16 +59,14 @@ impl KeySource {
                 use rusoto_ssm::Ssm;
 
                 let ssm_client = crate::ssm::build_client(profile.deref_shim())?;
-                let response = ssm_client
-                    .get_parameter(rusoto_ssm::GetParameterRequest {
-                        name: parameter_name.to_owned(),
-                        with_decryption: Some(true),
-                    })
-                    .sync()
-                    .context(error::SsmGetParameter {
-                        profile: profile.clone(),
-                        parameter_name,
-                    })?;
+                let response_fut = ssm_client.get_parameter(rusoto_ssm::GetParameterRequest {
+                    name: parameter_name.to_owned(),
+                    with_decryption: Some(true),
+                });
+                let response = block_on(response_fut).context(error::SsmGetParameter {
```

Comment:
I'm not super familiar with futures, but does this `block_on` call (or the underlying `ssm::GetParameterRequest`) have some kind of timeout?

Reply:
No, block_on will run forever, but the underlying libraries have http timeouts (I didn't look deeply into this, kind of assuming).